spacepy/irbempy/__init__.py

```diff
@@ -83,11 +83,2472 @@
+    >>> import spacepy.toolbox
+    >>> import numpy as np
+    >>> dosemod = spacepy.irbempy.Shieldose2()
+    >>> dosemod.set_shielding(depths=spacepy.toolbox.logspace(0.1, 15, 45), units='mm')
+    >>> e_spec = lambda E: 2*np.exp(-E/0.3)
+    >>> e_grid = spacepy.toolbox.logspace(0.01, 10, 50)
+    >>> dosemod.set_flux(e_spec(e_grid), e_grid, species='e')
+    >>> dosemod.get_dose(detector=10, nucmeth=3)
+    >>> import spacepy.plot
+    >>> spacepy.plot.style('spacepy')
+    >>> dosemod.plot_dose(source=['e'])
+
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.settings = dm.SpaceData()
+        """Settings for the dose calculation (`~spacepy.datamodel.SpaceData`).
+           Updated by `set_flux`, `set_shielding`, and `get_dose`."""
+        self.results = dm.SpaceData()
+        """Results of dose calculation from `get_dose`
+        (`~spacepy.datamodel.SpaceData`).
+
+        Keys are ``dose_proton_untrapped``,  ``dose_proton_trapped``,
+        ``dose_electron``, ``dose_bremsstrahlung``, ``dose_total``,
+        ``depths``, ``fluence_electron``.
+        """
+        self.settings['calc_flag'] = False
+        self.set_shielding()
+
+        def j_def(E, e_or_p='e'):
+            if e_or_p == 'e':
+                j = np.exp(-E/0.1)  # E in MeV
+            else:
+                j = 1e-9*np.exp(-E/20)
+            return j
+
+        en_e_default = tb.logspace(0.04, 10, 30)
+        en_p_default = tb.logspace(0.1, 2000, 299)
+        self.set_flux(j_def(en_e_default, 'e'), en_e_default, 'e')
+        self.set_flux(j_def(en_p_default, 'p'), en_p_default, 'p_tr')
+        self.set_flux(1e-6*j_def(en_p_default, 'p'), en_p_default, 'p_un')
+
+    def __repr__(self):
+        ndepths = len(self.settings['depths'])
+        calc = ' ' if self.settings['calc_flag'] else ' not '
+        settings = f'Depths = {ndepths}; Dose{calc}calculated'
+        return '<Shieldose2({})>'.format(settings)
+
+    def __str__(self):
+        '''
+        Display contents of container.
+        '''
+        kwargs = {'attrs': True, 'verbose': True,
+                  'spaces': '     ', 'print_out': False}
+        typestr = str(type(self)).split("'")[1]
+        settings = "|____settings\n" + self.settings.tree(**kwargs)
+        results = "|____results\n"\
+            + self.results.tree(**kwargs) if self.results else ""
+        return f"""{typestr}\n{settings}{results}"""
+
+    def set_shielding(self, depths=tb.logspace(4, 5000, 30),
+                      units='Mil'):
+        """
+        Parameters
+        ----------
+        depths : array-like
+            Array of layer depths (not thicknesses) in given units.
+            Must be monotonically increasing.
+
+        units : `str`
+            Options are "mil" (thousandths of a inch; default),
+            "g/cm2" (density), or "mm" (millimeters). The shielding
+            is assumed to be aluminium-equivalent.
+        """
```

Comment:
This is a good example of how docstrings should be formatted, look at how the 'd' in "depths" is exactly underneath the 'P' in "Parameters", this is what sphinx is looking for, especially important with bulleted or numbered lists